awscli/customizations/ec2instanceconnect/ssh.py

```diff
@@ -448,19 +442,23 @@ def _ssh(
 class KeyManager:
     def generate_key(self):
         logger.debug('Generate Ed25519 Key')
-        return Ed25519PrivateKey.generate()
+        return ED25519.new_generate()
+
+    def _get_public_key(self, private_key):
+        return f"ssh-ed25519 {ED25519.export_public_key(private_key, export_format=ED25519ExportFormat.OPENSSH_B64).decode()}"
```

Comment:
This should probably call `.encode()` on the resulting string to return bytes like `public_bytes` did.

Reply:
Fixed in cc20e49cb32730e13930c74cd03654e3c18b432d